memo/_base.py

```diff
@@ -1,6 +1,78 @@
 import orjson
-from typing import Callable, List
+from typing import Callable, Dict, List, Iterable
+from types import GeneratorType
 from functools import wraps
+from joblib import Parallel, delayed, parallel_backend
+import joblib.parallel
+from rich.progress import Progress
+import time
+import warnings
+
+
+class Runner():
```

Comment:
Let's add a docstring to the Runner class.

Reply:
Ok. This is the docstring I came up with for `Runner`. Let me know if there is anything  you'd like changed before I push it.

```python
    """Run functions in parallel with joblib
        joblib has 3 standard options for backends. 2 process based {"loky", "multiprocessing"}
        and 1 thread based {"threading"}. Depending on the function you wish to run one approach
        may be considerably faster than another. See links for more details

        The other important argument is the number of jobs `n_jobs` which dictates the
        number of cpu cores for proecess based backends or number of threads for thread
        based backend. You can pass -1 to use all cores available or the number
        you'd like to run e.g n_jobs=2. Be aware that over-subscription of CPU resources
        e.g n_jobs=32 on a 6 core processer will adversely affect performance
        See links for more details

        All keyword arguments during instantiaition will pass through to `parallel_backend`

        Notes
        joblib can also attach to third party backends such as Ray or Apache spark,
        however that functionality has not yet been tested

        Links
        https://joblib.readthedocs.io/en/latest/parallel.html
    """

```

Reply:
Just pushed the docstring